### ISSUE TITLE
Get-itglueconfigurations -filter_serial_number not all information is returned

### ISSUE THREAD

**Issue by bjowol** [itglue/powershellwrapper#51]
https://github.com/itglue/automation/issues/18

The same issue persists for not only Meraki devices. The problem is that the information that is displayed within itglue is not returned from the cmdlet. This is a huge bummer for since there is no good way to filter MAC devices from the rest of the bunch for example.

**Comment by adrianwells** [comment]
@bjowol, Hello. May you please test again with release v2.0.5? Testing this with version 2.0.5 returns valid information, an example cmdlet `(Get-ITGlueConfigurations -filter_serial_number 'ABC1239').data.attributes`

**Comment by bjowol** [comment]
![bilde](https://user-images.githubusercontent.com/14833255/51467765-9b8f2080-1d6d-11e9-934a-9832e9b2d513.png)

Not working here.

**Comment by adrianwells** [comment]
@bjowol - Thank you for the reply and screen shot. May you please provide an example of the cmdlet used to produce those results?

**Comment by bjowol** [comment]
![bilde](https://user-images.githubusercontent.com/14833255/52331100-8cf46a80-29f7-11e9-91b0-a71d17585026.png)

No problem.

I found another issue as well. regarding passwords. 

running this command:`(Get-ITGluePasswords -id 973936509305037 -show_password $true).data.attributes`
returns this output. Nothing wrong here, works as intended.
![bilde](https://user-images.githubusercontent.com/14833255/52331231-e6f53000-29f7-11e9-9edb-d729b9c6f943.png)

however, the problem is where you import all the passwords from an organization. 
`(Get-ITGluePasswords -filter_organization_id 'xxxxxxxxxxxxxx' -show_password $true).data.attrib
utes`

An error is returned
![bilde](https://user-images.githubusercontent.com/14833255/52333039-83213600-29fc-11e9-8f50-a7ca61aff94b.png)

So as of right now i cannot get all the password out via PS without doing a lot of api calls (3500).

**Comment by adrianwells** [comment]
@bjowol, thanks for the reply. Reviewing the report issue here and [itglue/automation#18 ](https://github.com/itglue/automation/issues/18), I think understand the issue and suggest engaging IT Glue Support if you have not already as it appears the data is not there. Both of the following examples return the same full data set from IT Glue when testing with version 2.0.7 after entering details like manufacturer name, id, etc, valid data is returned both snippets below return the same data:

```
(Get-ITGlueConfigurations -id 10844996).data.attributes
(Get-ITGlueConfigurations -filter_serial_number 'ABC1239').data.attributes
```
As for the password issue, may you please create a separate issue to break this out?

**Comment by bjowol** [comment]
i guess ill create a ticket. The data is available in the GUI

**Comment by bjowol** [comment]
> 
> 
> @bjowol, thanks for the reply. Reviewing the report issue here and [itglue/automation#18 ](https://github.com/itglue/automation/issues/18), I think understand the issue and suggest engaging IT Glue Support if you have not already as it appears the data is not there (sounds like this this part of the IT Glue Cisco Meraki intergration feature). Both of the following examples return the same full data set from IT Glue when testing with version 2.0.7 after entering details like manufacturer name, id, etc, valid data is returned both snippets below return the same data:
> 
> ```
> (Get-ITGlueConfigurations -id 10844996).data.attributes
> (Get-ITGlueConfigurations -filter_serial_number 'ABC1239').data.attributes
> ```
> 
> As for the password issue, may you please create a separate issue to break this out?

![bilde](https://user-images.githubusercontent.com/14833255/52567825-7ee68580-2e0d-11e9-81b1-1d0f03d30898.png)

The data is in itglue, but as you mentioned there is and integraiton feature issue. The "IT-Glue specific" field/object is not populated by the integration, as you can see here.

![bilde](https://user-images.githubusercontent.com/14833255/52567874-9887cd00-2e0d-11e9-8232-b8bb7db7df30.png)

Got the same issue with data coming from our PSA. The IP address of the device is visible in the GUI, in the IP Address field, but none of the data from the PSA is returned from the IT-Glue API. The information is just missing. I'll contact itglue support and get their input.

**Comment by bjowol** [comment]
Got this reply from it-glue support. looks like this module need an update. Have not gotten aroud to create a pull request, but i will look at it in the following months (got a lot to do before we start using ITG).

> @bjowol, thanks for the reply. Reviewing the report issue here and [itglue/automation#18 ](https://github.com/itglue/automation/issues/18), I think understand the issue and suggest engaging IT Glue Support if you have not already as it appears the data is not there (sounds like this this part of the IT Glue Cisco Meraki intergration feature). Both of the following examples return the same full data set from IT Glue when testing with version 2.0.7 after entering details like manufacturer name, id, etc, valid data is returned both snippets below return the same data:
> 
> ```
> (Get-ITGlueConfigurations -id 10844996).data.attributes
> (Get-ITGlueConfigurations -filter_serial_number 'ABC1239').data.attributes
> ```
> As for the password issue, may you please create a separate issue to break this out?

Thank you for your for your patience on the matter. The model and the rest of the RMM information can be retrieved with a GET request to: https://api.itglue.com/configurations/XXXXXXXX/?include=rmm_records

Sample output:

"included": [
        {
            "id": "XXXXXXXXX",
            "type": "rmm-records",
            "attributes": {
                "name": "XXXXXXXXXXX",
                "cpu": null,
                "device-type": "Switch",
                "external-url": null,
                "hostname": null,
                "last-login-user": null,
                "last-reboot": null,
                "last-seen-online": "2019-01-27T19:27:41.000Z",
                "manufacturer-name": "Meraki",
                "memory-total": null,
                "manufacturer-model-name": "MS220-8P",
                "ms-office-version": null,
                "patches-applied": null,
                "patches-total": null,
                "serial-number": "XXXX-XXXX-XXXX",
                "disks": null,
                "ip-addresses": [
                    "10.0.0.17"
                ],
                "operating-system": null,
                "mac-addresses": [
                    "88-XX-44-XX-ED-XX"

The  "manufacturer-model-name" will show up here.

With the PS Module, I was not able to see this option to hit the ?include=rmm_records option.

I ran: 
Get-Help GET-itglueconfigurations 
and could not see options that will ask for the RMM record data.

SYNTAX
    Get-ITGlueConfigurations [-organization_id <long>] [-filter_id <long>] [-filter_name <string>] [-filter_organization_id <long>]
    [-filter_configuration_type_id <long>] [-filter_configuration_status_id <long>] [-filter_contact_id <long>] [-filter_serial_number
    <string>] [-filter_rmm_id <string>] [-filter_rmm_integration_type {addigy | aem | atera | managed-workplace | continuum | jamf-pro |
    kaseya-vsa | automate | msp-rmm | msp-n-central | ninja-rmm | panorama9 | pulseway-rmm | watchman-monitoring}] [-sort {name | id |
    created_at | updated-at | -name | -id | -created_at | -updated-at}] [-page_number <long>] [-page_size <int>] [-include <string>]

Because we do not work on the wrapper, please contact the author to let them know you are trying to see this data.
_______________________________________________________________________________________________________________________________________

On the IT Glue side we have a feature request to have more RMM data write to IT Glue fields. 

Overwrite IT Glue data with RMM Data

I’ve added your vote to the idea on your behalf. This also subscribes you to alerts about the suggested feature. We encourage you to join the conversation by adding your comments. You’ll need to already be signed into IT Glue to view the idea.

If you have any other issues please feel free to respond.

Thank you for your engagement, and continued partnership with IT Glue.
Neil Shrestha-Birtch
Partner Success Manager, IT Glue
We ♥ Documentation™

**Comment by bjowol** [comment]
to get this up and running, add the marked line into the script "Configurations.ps1"

https://api.itglue.com/developer/
API documentation:

![image](https://user-images.githubusercontent.com/14833255/53732460-50911e80-3e7e-11e9-9daa-09a9ed378764.png)



![image](https://user-images.githubusercontent.com/14833255/53732441-3d7e4e80-3e7e-11e9-9d90-238bef87dbf0.png)
